### PR TITLE
Fix for time conversion Labs

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#9380](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9380)

* [ ] Enhancement
* [X] Bug
* [ ] Documentation

## Description
- Fix the time conversion for Labs
image

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [X] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested

* [X] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

**Review by sslaws — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts [hunk 1]
```diff
@@ -60,7 +60,11 @@ export class DateWrapper {
                     });
                     // DST times at 00:00 can be ambigious. Coherse the internal representation to be the day.
                     // 3 Hours bypases the ambiguousness of Move forward and fallback
-                    if (this._internal_date.isInDST) {
+                    if (
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts:63]: looks like the data is coming with timezone information. In that case I would check that and skip this section.

similar to:
```
if (param.includes("z")) {
                    this._internal_date = DateTime.fromISO(param);
                }

**Review by FuriousLlama — CHANGES_REQUESTED** [review]
Current change might cause other behaviour

File: Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts [hunk 2]
```diff
@@ -54,6 +54,10 @@ export class DateWrapper {
                     this._internal_date = DateTime.fromISO(param, {
                         zone: "utc",
                     }).setZone("America/Vancouver");
+                } else if (/-\d{2}:\d{2}$/.test(param)) {
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts:57]: this regex check if the param ends with -dd:dd, for example, "2020-10-25T10:56:52.1890766-07:00" param will return true, but "2020-10-22T00:00:00" returns false.

File: Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts]: @FuriousLlama I have updated the code, can you please check?

**Review by sslaws — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts:57]: what do you think about trying to capture the time zone? if that information is there then we can assume is not only a date, but a date time

the -07:00 part

Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts:57]: @FuriousLlama the "-07:00" part can't tell us which timezone right?

Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts:57]: Per my discussion with @FuriousLlama, I have removed the zone from the code above

**Review by sslaws — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**